object-store: Can't use `http` feature on its own

**Describe the bug**

With object store set as a dependency using:

```toml
object_store = {version = "0.7.0", optional = true, features = ["http"]}
```

I'm getting 

```
error[E0432]: unresolved import `crate::client::header`
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/object_store-0.7.0/src/http/mod.rs:43:20
   |
43 | use crate::client::header::header_meta;
   |                    ^^^^^^ could not find `header` in `client`

error[E0433]: failed to resolve: could not find `header` in `client`
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/object_store-0.7.0/src/http/mod.rs:66:32
   |
66 |         source: crate::client::header::Error,
   |                                ^^^^^^ could not find `header` in `client`
```

Specifically it looks like that mod is only built if one of aws, gcp, or azure are built.

https://github.com/apache/arrow-rs/blob/b66c57c4b441140f61af846a3371a957e82a5ff7/object_store/src/client/mod.rs#L38-L39

**To Reproduce**

Check out this PR: https://github.com/kylebarron/parquet-wasm/pull/311

```
> cargo build --features arrow1 --features async
```

```
error[E0432]: unresolved import `crate::client::header`
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/object_store-0.7.0/src/http/mod.rs:43:20
   |
43 | use crate::client::header::header_meta;
   |                    ^^^^^^ could not find `header` in `client`

   Compiling snap v1.1.0
error[E0433]: failed to resolve: could not find `header` in `client`
  --> /Users/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/object_store-0.7.0/src/http/mod.rs:66:32
   |
66 |         source: crate::client::header::Error,
   |                                ^^^^^^ could not find `header` in `client`
```

**Expected behavior**

I'm trying to use object store in wasm, so I can use only the `http` feature.

**Additional context**

--- a/.github/workflows/object_store.yml
+++ b/.github/workflows/object_store.yml
@@ -60,6 +60,8 @@ jobs:
         run: cargo clippy --features gcp -- -D warnings
       - name: Run clippy with azure feature
         run: cargo clippy --features azure -- -D warnings
+      - name: Run clippy with http feature
+        run: cargo clippy --features http -- -D warnings
       - name: Run clippy with all features
         run: cargo clippy --all-features -- -D warnings
       - name: Run clippy with all features and all targets

--- a/object_store/README.md
+++ b/object_store/README.md
@@ -39,7 +39,7 @@ See [docs.rs](https://docs.rs/object_store) for usage instructions
 
 ## Support for `wasm32-unknown-unknown` target
 
-It's possible to build `object_store` for the `wasm32-unknown-unknown` target, however the cloud storage features `aws`, `azure`, and `gcp` are not supported.
+It's possible to build `object_store` for the `wasm32-unknown-unknown` target, however the cloud storage features `aws`, `azure`, `gcp`, and `http` are not supported.
 
 ```
 cargo build -p object_store --target wasm32-unknown-unknown

--- a/object_store/src/client/mod.rs
+++ b/object_store/src/client/mod.rs
@@ -18,6 +18,7 @@
 //! Generic utilities reqwest based ObjectStore implementations
 
 pub mod backoff;
+
 #[cfg(test)]
 pub mod mock_server;
 
@@ -35,7 +36,6 @@ pub mod list;
 #[cfg(any(feature = "aws", feature = "gcp", feature = "azure"))]
 pub mod token;
 
-#[cfg(any(feature = "aws", feature = "gcp", feature = "azure"))]
 pub mod header;
 
 #[cfg(any(feature = "aws", feature = "gcp"))]
@@ -575,6 +575,7 @@ pub struct StaticCredentialProvider<T> {
 }
 
 impl<T> StaticCredentialProvider<T> {
+    /// A [`CredentialProvider`] for a static credential of type `T`
     pub fn new(credential: T) -> Self {
         Self {
             credential: Arc::new(credential),

--- a/object_store/src/lib.rs
+++ b/object_store/src/lib.rs
@@ -240,9 +240,9 @@
 
 #[cfg(all(
     target_arch = "wasm32",
-    any(feature = "gcp", feature = "aws", feature = "azure",)
+    any(feature = "gcp", feature = "aws", feature = "azure", feature = "http")
 ))]
-compile_error!("Features 'gcp', 'aws', 'azure' are not supported on wasm.");
+compile_error!("Features 'gcp', 'aws', 'azure', 'http' are not supported on wasm.");
 
 #[cfg(feature = "aws")]
 pub mod aws;
@@ -263,13 +263,16 @@ pub mod path;
 pub mod prefix;
 pub mod throttle;
 
-#[cfg(any(feature = "gcp", feature = "aws", feature = "azure", feature = "http"))]
+#[cfg(feature = "cloud")]
 mod client;
 
-#[cfg(any(feature = "gcp", feature = "aws", feature = "azure", feature = "http"))]
-pub use client::{backoff::BackoffConfig, retry::RetryConfig, CredentialProvider};
+#[cfg(feature = "cloud")]
+pub use client::{
+    backoff::BackoffConfig, retry::RetryConfig, ClientConfigKey, ClientOptions,
+    CredentialProvider, StaticCredentialProvider,
+};
 
-#[cfg(any(feature = "gcp", feature = "aws", feature = "azure", feature = "http"))]
+#[cfg(feature = "cloud")]
 mod config;
 
 #[cfg(feature = "cloud")]
@@ -294,9 +297,6 @@ use std::io::{Read, Seek, SeekFrom};
 use std::ops::Range;
 use std::sync::Arc;
 use tokio::io::AsyncWrite;
-
-#[cfg(any(feature = "azure", feature = "aws", feature = "gcp", feature = "http"))]
-pub use client::{ClientConfigKey, ClientOptions};
 
 /// An alias for a dynamically dispatched object store implementation.
 pub type DynObjectStore = dyn ObjectStore;

--- a/object_store/src/parse.rs
+++ b/object_store/src/parse.rs
@@ -104,7 +104,7 @@ impl ObjectStoreScheme {
     }
 }
 
-#[cfg(any(feature = "aws", feature = "gcp", feature = "azure", feature = "http"))]
+#[cfg(any(feature = "aws", feature = "gcp", feature = "azure"))]
 macro_rules! builder_opts {
     ($builder:ty, $url:expr, $options:expr) => {{
         let builder = $options.into_iter().fold(

